libraries/Wire/src/Wire.h

```diff
@@ -81,12 +83,42 @@ class TwoWire: public Stream
     static void onReceiveService(uint8_t, uint8_t*, size_t, bool, void *);
 #endif /* SOC_I2C_SUPPORT_SLAVE */
     bool initPins(int sdaPin, int sclPin);
-    bool allocateWireBuffer(void);
-    void freeWireBuffer(void);
+    bool allocateWireBuffer();
+    void freeWireBuffer();
 
 public:
     TwoWire(uint8_t bus_num);
     ~TwoWire();
+
+    bool begin() override final
+    {
+        return begin(-1, -1);
+    }
+
+    bool begin(uint8_t address) override final
+    {
+#if SOC_I2C_SUPPORT_SLAVE
+        return begin(address, -1, -1, 0);
+#else
+        log_e("I2C slave is not supported by you SoC!!!");
+#endif
+    }
```

Comment:
CI is failing here as no return statement if slave is not supported.
```suggestion
    bool begin(uint8_t address) override final
    {
#if SOC_I2C_SUPPORT_SLAVE
        return begin(address, -1, -1, 0);
#else
        log_e("I2C slave is not supported on " CONFIG_IDF_TARGET);
        return false;
#endif
    }
```